pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -267,13 +259,9 @@ int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
     return 0;
 }
 
-int sock_dtls_session_create(sock_dtls_t *sock, const sock_udp_ep_t *ep,
-                             sock_dtls_session_t *remote)
+int sock_dtls_session_init(sock_dtls_t *sock, const sock_udp_ep_t *ep,
+                           sock_dtls_session_t *remote)
 {
-    uint8_t rcv_buffer[DTLS_HANDSHAKE_BUFSIZE];
-    msg_t msg;
-    ssize_t res;
-
     assert(sock);
     assert(ep);
     assert(remote);
```

Comment:
This way, your code is up to the spec:

```suggestion
    assert(remote);
    sock_udp_ep_t local;
    
    if (sock_udp_get_local(sock->udp_sock, &local) < 0) {
        return -EADDRUNAVAIL;
    }
    if (ep->port == 0) {
        return -EINVAL;
    }
    switch (ep->family) {
        case AF_INET:
 #if IS_ACTIVE(SOCK_HAS_IPV6)
        case AF_INET6:
 #endif
            break;
        default:
            return -EINVAL;
    }
```

(did not compile test this one... hope you know what I mean)

Reply:
not sure, if `sock->udp_sock` should also be checked to be set first as well.